contrib/babelfishpg_tsql/src/rolecmds.c

```diff
@@ -1254,7 +1257,10 @@ add_to_bbf_authid_user_ext(const char *user_name,
 	else
 		new_record_user_ext[USER_EXT_TYPE] = CStringGetTextDatum("S");
 	new_record_user_ext[USER_EXT_OWNING_PRINCIPAL_ID] = Int32GetDatum(-1);	/* placeholder */
-	new_record_user_ext[USER_EXT_IS_FIXED_ROLE] = Int32GetDatum(-1);	/* placeholder */
+	if (strcmp(orig_user_name, DBO) != 0 && IS_FIXED_DB_PRINCIPAL(orig_user_name))
```

Comment:
Question: Why `IS_FIXED_DB_PRINCIPAL` marks dbo as a fixed role if it is not actually a fixed role from TSQL perspective?

Reply:
Ideally `dbo` is a  fixed role and `IS_FIXED_DB_PRINCIPAL` is an internal MACRO used at multiple places where the functionality matches for `dbo` and other database roles. So, lets keep it that way.
But in sys.database_principals view, TSQL doesn't show `dbo` as a fixed role.